### ISSUE TITLE
graphql() false positive

### ISSUE THREAD

**Issue by developer239** [eslint/eslint#8171]
**Tell us about your environment**

* **ESLint Version: "^3.15.0"**
* **Node Version: "v7.5.0"**
* **npm Version: "4.1.2"**

**What parser (default, Babel-ESLint, etc.) are you using?**

"webpack": "^2.2.1",
"babel-eslint": "^7.1.1",

**What did you do? Please include the actual source code causing the issue.**

```js

import React from 'react'
import { Link } from 'react-router'
import { graphql } from 'react-apollo'
import { connect } from 'react-redux'
import { Grid, Alert, PageHeader } from 'react-bootstrap'

import Spinner from 'components/common/Spinner'
import { queryPostDetail } from '../../qql'


export const PostsBaseContainer = ({ data }) => {
  if (data.loading) {
    return (
      <Grid>
        <Spinner />
      </Grid>
    )
  }

  if (data.error) {
    return (
      <Grid>
        <Alert bsStyle="danger">
          <strong>Error!</strong> Fetching data from GraphQl server was not successful.
        </Alert>
      </Grid>
    )
  }

  return (
    <Grid>
      <PageHeader>
        {data.Post.title}
        <Link to={`/posts/${data.Post.id}/edit`}>
          <small>[edit]</small>
        </Link>
      </PageHeader>
      <p>{data.Post.text}</p>
    </Grid>
  )
}

PostsBaseContainer.propTypes = {
  data: React.PropTypes.shape({
    loading: React.PropTypes.bool,
    error: React.PropTypes.object,
    Post: React.PropTypes.shape({
      id: React.PropTypes.string.isRequired,
      title: React.PropTypes.string.isRequired,
      text: React.PropTypes.string.isRequired,
    }),
  }).isRequired,
  params: React.PropTypes.shape({
    postId: React.PropTypes.string.isRequired,
  }).isRequired,
}

const mapStateToProps = () => ({})
const mapDispatchToProps = () => ({})

const PostsBaseContainerWithQueries = graphql(queryPostDetail, {
  options: ownProps => ({
    variables: {
      id: ownProps.params.postId,
    },
  }),
})(PostsBaseContainer)


export default connect(mapStateToProps, mapDispatchToProps)(PostsBaseContainerWithQueries)




```

**What did you expect to happen?**

Eslint should not find any errors in the code.

**What actually happened? Please include the actual, raw output from ESLint.**

![image](https://cloud.githubusercontent.com/assets/14946081/23458714/a0428ef2-fe7d-11e6-8dc8-d3fe9b48568d.png)

**Comment by not-an-aardvark** [comment]
It looks like this problem is being reported by the `no-unused-prop-types` rule from `eslint-plugin-react`, which is not maintained by the ESLint team. I would recommend filing an issue on [eslint-plugin-react](https://github.com/yannickcr/eslint-plugin-react/issues) instead.

**Comment by not-an-aardvark** [comment]
Closing this since it's not an issue with ESLint.
